sui_core/src/gateway_state.rs

```diff
@@ -809,10 +810,16 @@ where
     async fn get_transaction(
         &self,
         digest: TransactionDigest,
-    ) -> Result<CertifiedTransaction, anyhow::Error> {
+    ) -> Result<TransactionResponse, anyhow::Error> {
         let opt = self.store.get_certified_transaction(&digest)?;
         match opt {
-            Some(t) => Ok(t),
+            Some(t) => {
+                let effect = self
+                    .authorities
+                    .process_certificate(t.clone(), AUTHORITY_REQUEST_TIMEOUT)
+                    .await?;
+                Ok(TransactionResponse::EffectResponse(t, effect))
+            }
```

Comment:
The effects should exist in the store (in the `effects` table). You can add a function to expose that

Reply:
I tried to read from the gateway's effects table but it's always empty.... is that expected?

I am checking the gateway code to see if we are persisting the effects or not.

Reply:
I have added persisting logic for transaction effects in gateway

https://github.com/MystenLabs/sui/blob/6f5b33b6e6e410de885b9cc62b4c9367245aac98/sui_core/src/authority/authority_store.rs#L618-L622

Reply:
Hmm that's surprising to me. I am away from laptop atm. Will take a closer look tonight.